### DISCUSSION TITLE
How pass a record into formatStateUsing method  ?

### DISCUSSION THREAD

**Discussion by PetroGromovo** [filamentphp/filament#9787] [Help] [answered]
When in a table I need to show formatted value for some column I use formatStateUsing method:
```
                TextColumn::make('created_at')
                    ->formatStateUsing(fn(string $state): string => Helper::getDateTimeFormater($state))
                    ...

```
But I did not find how can I pass a record itself, as resulting value is calculated value based on several field :

```
                TextColumn::make('summary_points_percent')
                    ->formatStateUsing(fn(?Item $record) => Helper::getSummaryPointsPercent($record))

```
Method getSummaryPointsPercent - returns string, but is is not rendered, as I see empty cell field...

```
"laravel/framework": "^10.28.0",
"filament/filament": "^3.0-stable",

```
Thanks in advance!

**Comment by leandrocfe** [comment]
Is the $record null?
`->formatStateUsing(fn(?Item $record) => $record->field) //null`

**Reply by PetroGromovo** [reply]
In my table there are several fields I show :
```
return $table
    ->query(Item::query()->getByStatus(true))
    ->striped()
    ->columns([
        TextColumn::make('id')->sortable(),
        TextColumn::make('user_name')->sortable()->searchable(), // I see this field
        TextColumn::make('user_email')->sortable()->searchable(),

        TextColumn::make('summary_points_percent')
//                    ->formatStateUsing(fn(string $state): string => ItemHelper::getSummaryPointsPercent(Item))
            ->formatStateUsing(fn(?Item $record) => $record->user_name) // I DO NOT SEE THIS FIELD - but I see it with TextColumn

```
I suppose I need to use other method instead of formatStateUsing or define $record in some other way ?

**Reply by leandrocfe** [reply]
try `getStateUsing()`

**Reply by PetroGromovo** [reply]
I docs I did not any reference to getStateUsing...

**Reply by leandrocfe** [reply, marked as answer]
`TextColumn::make('summary_points_percent')->getStateUsing(fn($record) => $record->user_name)`
What is the output?

**Comment by eloi-charly01** [comment]
Try like that
`
 Tables\Columns\TextColumn::make('valeur_ht')
                ->label('Valeur HT en €')
                ->formatStateUsing(function ($record): string {

                    if ($record->valeur_ht === 'product') {
                        return DebCreditMemoRepository::getRowTotalProduct($record->creditmemo_number);
                    }

                    return $record->valeur_ht;
                })
                ->toggleable(),
`

**Comment by leandrocfe** [comment]
We can use `state` now
https://filamentphp.com/docs/3.x/tables/columns/getting-started/#calculated-state

**Reply by djl997** [reply]
How to do this if the TextEntry is in a RepeatableEntry? I cannot get this to work... (in a infolist)

```php
Infolists\Components\RepeatableEntry::make('products')->schema([
    Infolists\Components\TextEntry::make('value')
        ->money('EUR'),
    Infolists\Components\TextEntry::make('quantity'),
    Infolists\Components\TextEntry::make('total')
        ->money('EUR')
        ->state(function($record) {
            // returns record, but not the repeated row??
        }),
])
```

**Reply by leandrocfe** [reply]
I think `$record` is the whole Record, not the products. Open a new topic and share the whole code you are using

**Reply by djl997** [reply]
Yup. 
Made a new topic: https://github.com/filamentphp/filament/discussions/14773
